bundles/org.eclipse.text/src/org/eclipse/jface/text/ListLineTracker.java

```diff
@@ -174,8 +174,10 @@ public final IRegion getLineInformationOfOffset(int position) throws BadLocation
 	public final IRegion getLineInformation(int line) throws BadLocationException {
 		int lines= fLines.size();
 
-		if (line < 0 || line > lines)
-			throw new BadLocationException();
+		if (line > lines)
+			throw new BadLocationException("line > lines: " + line + " > " + lines);  //$NON-NLS-1$//$NON-NLS-2$
+		if (line < 0)
+			throw new BadLocationException("line < 0: " +line);  //$NON-NLS-1$
```

Comment:
In such case I think something like
```suggestion
			throw new BadLocationException("Negative values are not allowed for parameter line..");  //$NON-NLS-1$
```